.github/workflows/test.yml

```
@@ -28,66 +34,44 @@ jobs:
           echo $IMAGE_URL
 
   build-and-check:
-    name: Building, testing, linting
+    name: Build, test, lint
     needs: extract-image
     runs-on: self-hosted
-    concurrency:
-      group: build-and-check-${{ github.ref }}
-      cancel-in-progress: true
-    env:
-      CMAKE_ARGS: "-DCMAKE_EXPORT_COMPILE_COMMANDS=1 -DCMAKE_TOOLS_ADDRESS_SANITIZER=1"
-
     container:
       image: ${{ needs.extract-image.outputs.image-url}}
 
+    # TODO: Fix in dockerfile
+    defaults:
+      run:
+        shell: bash
+
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v3
+      - uses: actions/checkout@v3
 
       - name: Build ROS2 packages
-        shell: bash
         run: |
           source $ROS_ROOT/setup.bash
           cd $GITHUB_WORKSPACE/packages
-          colcon build --merge-install --cmake-args $CMAKE_ARGS
+          colcon build --merge-install --cmake-args \
+            -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
+            -DCMAKE_TOOLS_ADDRESS_SANITIZER=ON
           source install/setup.bash
 
       - name: Test ROS2 packages
         continue-on-error: true
-        shell: bash
         run: |
           cd $GITHUB_WORKSPACE/packages
-          colcon test --ctest-args tests --merge-install --executor parallel --parallel-workers $(nproc) --return-code-on-test-failure
+          colcon test --ctest-args tests --merge-install \
+            --executor parallel --parallel-workers $(nproc) \
+            --return-code-on-test-failure
 
       - name: Run Clang-Tidy
-        shell: bash
         run: |
-          FILES=$(find . \( -name '*.h' -or -name '*.cpp' -or -name '*.cc' \) -not -path '*/build/*' -not -path '*/install/*' -not -path '*/log/*')
+          FILES=$(find . \( -name '*.h' -or -name '*.cpp' -or -name '*.cc' \) \
+            -not -path '*/build/*' -not -path '*/install/*' -not -path '*/log/*')
           echo "Files to be linted:"
           echo "$FILES"
           for FILE in $FILES; do
             echo $FILE
             clang-tidy --fix -p=packages/build $FILE
           done
```

Будем ли убирать цикл (тебе он не нравился)?
И еще была идея все же взять тот `.py` [скрипт](https://github.com/vmware/concord-bft/blob/61c4d39e5ca71690817e341c9713e014b91a13d8/tools/run-clang-tidy.py)
```suggestion
          echo "Files to be linted:"
          echo $FILES
          clang-tidy --fix -p=packages/build $FILEЫ
```

---

Да, надо еще подумать насчет этого. Пока что есть желание все в makefile положить для начала